woocommerce/woocommerce
# Migrate `@woocommerce/data` item store to TS

### All Submissions:






### Changes proposed in this Pull Request:



Part of https://github.com/woocommerce/woocommerce/issues/33105

This PR migrates item store to TS and fixes a wrong type annotation in ./client after migrating onboarding store to TS

### How to test the changes in this Pull Request:

1. Should run `pnpm -- turbo run build --filter=@woocommerce/data` without errors
2. Should run `pnpm run --filter woocommerce/client/admin ts:check` without errors

### Other information:




### FOR PR REVIEWER ONLY:
